fineract-provider/src/main/java/org/apache/fineract/portfolio/interestratechart/data/InterestRateChartSlabDataValidator.java

```diff
@@ -239,7 +239,7 @@ public void validateChartSlabsUpdate(final JsonElement element, final DataValida
         }
 
         if (amountRangeFrom != null && amountRangeTo != null) {
-            if (amountRangeFrom.compareTo(amountRangeTo) > 1) {
+            if (amountRangeFrom.compareTo(amountRangeTo) > 0) {
```

Comment:
Just curious of this change, @percyashu can you please tell why they mean the same thing?

Reply:
Always comparing to 0 is the safest use of the return value when using compareTo since the magnitude of the value return is not certain but the sign is. See https://errorprone.info/bugpattern/CompareToZero

Reply:
This is actually "fun"... I also had to read https://errorprone.info/bugpattern/CompareToZero - and now I understand it. What's really interesting and not just fun is that in N places of this PR you're just doing something to make EP happy, but in exactly 4 occurrences, you've actually just fixed real bugs - see FINERACT-1014; thank You, Error Prone!

Reply:
@vorburger @percyashu this is interesting!